tests/mssql_test.go

```diff
@@ -31,78 +66,286 @@ func TestMain(m *testing.M) {
+					foundSampleTypes := make(map[string]bool)
+
+					for _, sample := range samples {
+						sample = strings.TrimSpace(sample)
+						if sample == "" {
+							continue
+						}
+
+						sampleType := getSampleType(sample, allSampleTypes)
+						require.NotEmpty(t, sampleType, "No sample type found in JSON output: %s", sample)
+						foundSampleTypes[sampleType] = true
+
+						validateSample(t, sample, sampleType)
+					}
+
+					verifySampleTypes(t, tt.expectedSampleTypes, foundSampleTypes)
+				})
+			}
 		})
-		if err != nil {
-			log.Warn(err.Error())
-
-			mssql_command := []string{"docker", "logs", "mssql"}
-			mssql_cmd := exec.Command(mssql_command[0], mssql_command[1:]...)
-			var mssql_out bytes.Buffer
-			mssql_cmd.Stdout = &mssql_out
-			mssql_cmd.Stderr = &mssql_out
-
-			nri_mssql_command := []string{"docker", "logs", "nri-mssql"}
-			nri_mssql_cmd := exec.Command(nri_mssql_command[0], nri_mssql_command[1:]...)
-			var nri_mssql_out bytes.Buffer
-			nri_mssql_cmd.Stdout = &nri_mssql_out
-			nri_mssql_cmd.Stderr = &nri_mssql_out
-
-			log.Info("mssql container logs")
-			_ = mssql_cmd.Run()
-			log.Info(mssql_out.String())
-
-			log.Info("nri_mssql container logs")
-			_ = nri_mssql_cmd.Run()
-			log.Info(nri_mssql_out.String())
-
-			continue
+	}
+}
+
+func runQueries(t *testing.T, container string) {
+	t.Helper()
+	containerPort, containerDB := getPortAndDbForContainer(container)
+	for i := 0; i < querySimCount; i++ {
+		err := simulation.SimulateDBQueries(t, containerPort, containerDB, *user, *psw)
+		require.NoError(t, err, "Failed to simulate database queries")
+	}
+}
+
+func validateSample(t *testing.T, sample string, sampleType string) {
+	t.Helper()
+	// Validate JSON format
+	var j map[string]interface{}
+	err := json.Unmarshal([]byte(sample), &j)
+	require.NoError(t, err, "Got an invalid JSON as output: %s", sample)
+
+	// Validate schema
+	t.Run(fmt.Sprintf("Validating JSON schema for sample: %s", sampleType), func(t *testing.T) {
+		schemaFile := getSchemaFileName(sampleType)
+		require.NotEmpty(t, schemaFile, "Schema file not found for sample type: %s", sampleType)
+
+		err = validateJSONSchema(schemaFile, sample)
+		assert.NoError(t, err, "Sample failed schema validation for type: %s", sampleType)
+	})
+}
+
+func verifySampleTypes(t *testing.T, expectedTypes []string, foundTypes map[string]bool) {
+	t.Helper()
+
+	// Check that all expected sample types were found
+	for _, sampleType := range allSampleTypes {
+		if slices.Contains(expectedTypes, sampleType) {
+			assert.True(t, foundTypes[sampleType],
+				"Expected sample type %s was not found in the output", sampleType)
 		}
-		if conn != nil {
-			conn.Close()
-			log.Info("mssql is up & running!")
-			return true
+	}
+
+	// Check that no unexpected sample types were found
+	for foundType := range foundTypes {
+		assert.True(t, slices.Contains(expectedTypes, foundType),
+			"Found unexpected sample type %s in the output", foundType)
+	}
+}
```

Comment:
Why iterating `allSampleTypes`? It is not needed, is it?

You may also simplify using [assert.Contains](https://pkg.go.dev/github.com/stretchr/testify/assert#Contains)